src/main/java/org/opensearch/knn/engine/method/HnswContext.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.engine.method;
+
+import com.google.common.collect.ImmutableMap;
+import org.opensearch.knn.index.Parameter;
+import org.opensearch.knn.index.query.request.MethodParameter;
+
+import java.util.Map;
+
+public class HnswContext implements EngineSpecificMethodContext {
```

Comment:
So is idea for engine to implement HnswCOntext? Im not sure I understand how HnswContext is EngineSpecificMethodContext.

Reply:
> So is idea for engine to implement HnswCOntext

Not really. See [Faiss](https://github.com/opensearch-project/k-NN/pull/1707/files#diff-019b2da767e6ac2fa27b70e74d14a35936c7546562aeecd3fac7703142bd4405R338) and [doToQuery](https://github.com/opensearch-project/k-NN/pull/1707/files#diff-6d0df17543181185170f65c116cdb3514d910e1557f012b7959611ae92678080R512) validation for usage.

> Im not sure I understand how HnswContext is EngineSpecificMethodContext.

I understand the confusion. I had initially kept this FaissHnswContext. But then thinking about nmslib/lucene, considering there is no difference currently I wanted a way to reuse and let it have engine specific stuff when required. I have a hunch we will have to split it once we try and add KNNMethod in here. I will add a comment and change it to DefaultHnswContext. Thoughts?